finesse/hardware/dummy_stepper_motor.py

```diff
@@ -0,0 +1,52 @@
+"""Code for a fake stepper motor device."""
+import logging
+from typing import Union
+
+from pubsub import pub
+
+_PRESETS = ("zenith", "nadir", "hot_bb", "cold_bb", "home")
+
+
+class DummyStepperMotor:
+    """A fake stepper motor device used for unit tests etc."""
+
+    def __init__(self, steps_per_rotation: int) -> None:
+        """Create a new DummyStepperMotor.
+
+        Args:
+            steps_per_rotation: Number of motor steps for an entire rotation (360°)
+        """
+        if steps_per_rotation < len(_PRESETS):
+            raise ValueError(f"steps_per_rotation must be >={len(_PRESETS)}")
+
+        self.steps_per_rotation = steps_per_rotation
+        self.current_step = 0
+
+        pub.subscribe(self.move_to, "stepper.move")
+
+    @staticmethod
+    def get_preset_step(name: str) -> int:
+        """Get the angle for one of the preset positions.
+
+        Args:
+            name: Name of preset angle
+        Returns:
+            The step number (as int)
+        """
+        return _PRESETS.index(name)
+
+    def move_to(self, target: Union[float, str]) -> None:
+        """Move the motor to a specified rotation.
+
+        Args:
+            target: The target angle (in degrees) or the name of a preset
+        """
+        if isinstance(target, str):
+            self.current_step = DummyStepperMotor.get_preset_step(target)
+        else:
+            step = round(self.steps_per_rotation * target / 360.0)
+            if step < 0 or step >= self.steps_per_rotation:
+                raise ValueError("step ")
```

Comment:
So we need something more informative here?

Reply:
Oops! Yes we do